waspc/cli/Command/Compile.hs

```diff
@@ -0,0 +1,35 @@
+module Command.Compile
+    ( compile
+    ) where
+
+import           Control.Monad.IO.Class (liftIO)
+import           Data.List              (find, isSuffixOf)
+import qualified Path                   as P
+
+import qualified Common
+import           CompileOptions         (CompileOptions (..))
+import qualified Lib
+import           StrongPath             (Abs, Dir, Path, (</>))
+import qualified StrongPath             as SP
+import qualified Util.IO
+
+
+-- | Compiles Wasp source code in waspRoot directory and generates a project in given outDir directory.
+compile :: Path Abs (Dir Common.WaspProjectDir) -> Path Abs (Dir Lib.ProjectRootDir) -> IO (Either String ())
+compile waspRoot outDir = do
+    maybeWaspFile <- findWaspFile waspRoot
+    case maybeWaspFile of
+        Nothing -> return $ Left "No .wasp file present in the root of Wasp project."
+        Just waspFile -> Lib.compile waspFile outDir options
+  where
+    options = CompileOptions
+        { externalCodeDirPath = waspRoot </> Common.extCodeDirInWaspProjectDir }
```

Comment:
ok so waspRoot is obviously WaspProjectDir. Maybe we should use just one name for that? Wasp project dir sounds more explanatory to me.

Reply:
Yes, done!